src/engine/plugins/trip.cpp

```diff
@@ -117,24 +117,40 @@ SCC_Component SplitUnaccessibleLocations(const std::size_t number_of_locations,
 InternalRouteResult
 TripPlugin::ComputeRoute(const std::shared_ptr<const datafacade::BaseDataFacade> facade,
                          const std::vector<PhantomNode> &snapped_phantoms,
-                         const std::vector<NodeID> &trip) const
+                         const std::vector<NodeID> &trip,
+                         const bool roundtrip) const
 {
     InternalRouteResult min_route;
-    // given he final trip, compute total duration and return the route and location permutation
+    // given the final trip, compute total duration and return the route and location permutation
     PhantomNodes viapoint;
     const auto start = std::begin(trip);
     const auto end = std::end(trip);
     // computes a roundtrip from the nodes in trip
     for (auto it = start; it != end; ++it)
     {
         const auto from_node = *it;
+
+        // if from_node is the last node and it is a fixed start and end trip,
+        // break out of this loop and return the route
+        if (!roundtrip && std::next(it) == end)
+            break;
+
         // if from_node is the last node, compute the route from the last to the first location
         const auto to_node = std::next(it) != end ? *std::next(it) : *start;
 
         viapoint = PhantomNodes{snapped_phantoms[from_node], snapped_phantoms[to_node]};
         min_route.segment_end_coordinates.emplace_back(viapoint);
     }
-    BOOST_ASSERT(min_route.segment_end_coordinates.size() == trip.size());
+
+    if (roundtrip)
+    {
+        BOOST_ASSERT(min_route.segment_end_coordinates.size() == trip.size());
+    }
+    else
+    {
+        // trip comes out to be something like 0 1 4 3 2, so the sizes don't
+        BOOST_ASSERT(min_route.segment_end_coordinates.size() == trip.size() - 1);
+    }
```

Comment:
I did this, but it got a bit confusing for `&&` / `||` noobs like me to keep the logic straight. So I reverted back to the if statements for clarity and simplicity's sake.